src/components/ListItem.jsx

```diff
@@ -1,5 +1,37 @@
 import './ListItem.css';
+import { updateItem } from '../api';
+import { sub } from 'date-fns';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const handleUpdate = async (isChecked) => {
+		let day = Date.now();
+		let purchaseCounter = item.totalPurchases + 1;
+
+		try {
+			if (isChecked) {
+				const newData = { day, purchaseCounter };
+				await updateItem(listToken, item.id, newData);
+				console.log('success');
+			}
+		} catch (error) {
+			console.error(error);
+		}
+	};
+
+	return (
+		<li className="ListItem">
+			<label htmlFor={item.id}>
+				<input
+					type="checkbox"
+					id={item.id}
+					defaultChecked={sub(new Date(), { days: 1 }) < item.dateLastPurchased}
```

Comment:
💡 Instead of doing this twice you could store the result of calling this function into a variable, e.g. isRecentlyPurchased, and then you could simplify your props like this: 

```
checked={!isRecentlyPurchased}
disabled={isRecentlyPurchased}
```

Reply:
Done.